pennylane_lightning/core/src/simulators/lightning_kokkos/utils/UtilKokkos.hpp

```diff
@@ -115,4 +116,123 @@ inline auto wires2Parity(const std::size_t num_qubits,
     return {parity, rev_wire_shifts};
 }
 
+/**
+ * @brief Compute parity and reverse wires for multi-qubit control operations
+ *
+ * @param num_qubits Number of qubits in the state vector.
+ * @param wires List of target wires.
+ * @param controlled_wires List of control wires.
+ * @return std::pair<KokkosIntVector, KokkosIntVector> Parities and reverse
+ * wires for control multi-qubit operations
+ */
+inline auto reverseWires(const std::size_t num_qubits,
+                         const std::vector<std::size_t> &wires,
+                         const std::vector<std::size_t> &controlled_wires)
+    -> std::pair<KokkosIntVector, KokkosIntVector> {
+    KokkosIntVector parity;
+    KokkosIntVector rev_wires;
+
+    const std::size_t n_contr = controlled_wires.size();
+    const std::size_t n_wires = wires.size();
+    const std::size_t nw_tot = n_contr + n_wires;
+    std::vector<std::size_t> all_wires;
+    all_wires.reserve(nw_tot);
+    all_wires.insert(all_wires.begin(), wires.begin(), wires.end());
+    all_wires.insert(all_wires.begin() + wires.size(), controlled_wires.begin(),
+                     controlled_wires.end());
+
+    std::vector<std::size_t> rev_wires_(nw_tot);
+    for (std::size_t k = 0; k < nw_tot; k++) {
+        rev_wires_[k] = (num_qubits - 1) - all_wires[(nw_tot - 1) - k];
+    }
+    const std::vector<std::size_t> parity_ = revWireParity(rev_wires_);
+
+    Kokkos::View<const std::size_t *, Kokkos::HostSpace,
+                 Kokkos::MemoryTraits<Kokkos::Unmanaged>>
+        rev_wires_host(rev_wires_.data(), rev_wires_.size());
+    Kokkos::resize(rev_wires, rev_wires_host.size());
```

Comment:
This is necessary because `rev_wires` (Kokkos view vector) does not have a size yet, only the host vector (`rev_wires_`) and the host view of the same vector (`rev_wires_host`), and this copy fails when they are not the same size.